File: Api/Responses/Game/RoundEndResponse.cs
using OhMyWord.Core.Models;

namespace OhMyWord.Api.Responses.Game;

public class RoundEndResponse
{
    public int RoundNumber { get; init; }
    public Guid RoundId { get; init; }
    public string Word { get; init; } = string.Empty;
    public RoundEndReason EndReason { get; init; }
    public DateTime NextRoundStart { get; init; }
}

File: Client/src/app/models/responses/round-end.response.ts
export interface RoundEndResponse {
  roundNumber: number;
  roundId: string;
  word: string;
  endReason: RoundEndReason;
  nextRoundStart: string;
}

export enum RoundEndReason {
  Timeout = 'timeout',
  AllPlayersAwarded = 'allPlayersAwarded',
  NoPlayersLeft = 'noPlayersLeft',
}
